gitlab/gitlab_test.go

```diff
@@ -43,3 +50,107 @@ func TestGetProjectID(t *testing.T) {
 		t.Errorf("Expected %s, got %s", "1", res)
 	}
 }
+
+func TestGetProjectIDwithNonexistentProject(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+	mockURL := "https://" + "gitlab.com" + "/api/v4"
+	httpmock.RegisterResponder("GET", "https://gitlab.com/api/v4/projects/",
+		httpmock.NewStringResponder(404, ""))
+	_, err := GetProjectID(mockURL, "213123", "test", "test")
+	if err == nil {
+		t.Errorf("Expected to get an error when project does not exist")
+	}
+}
+
+func TestGitlabCreateAndDisplayNewMilestones(t *testing.T) {
+	milestoneData, err := utils.CreateMilestoneData(10, "daily", nil, "gitlab")
+	if err != nil {
+		t.Error(err)
+	}
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+	mockURL := "https://" + "gitlab.com" + "/api/v4"
+	MockGitlabAPIGetRequest(mockURL, "active")
+	MockGitlabAPIPostRequest(mockURL, "active")
+	err = CreateAndDisplayNewMilestones(mockURL, "213123", "1", milestoneData, logger)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestPaginate(t *testing.T) {
```

Comment:
Maybe this function isn't needed twice, as the code is identical to the one in `github_test.go` (or I missed a difference?)

Reply:
Thanks! Since the GitLab functions need to use a []gitlabAPI, I don't think it would be possible to use those GitHub functions unfortunately.

Reply:
Thanks - I'm removing these duplicate functions. And I think this duplicate test is not needed either https://github.com/dbriesz/gomiler/blob/9c9a4877cda693f68fdd978a3fad9a2540ad587d/gitlab/gitlab_test.go#L95.